libcontainer/nsenter/cloned_binary.c

```diff
@@ -59,14 +59,38 @@
 #include <sys/syscall.h>
 
 /* Use our own wrapper for memfd_create. */
-#if !defined(SYS_memfd_create) && defined(__NR_memfd_create)
-#  define SYS_memfd_create __NR_memfd_create
+#ifndef SYS_memfd_create
+#  ifdef __NR_memfd_create
+#    define SYS_memfd_create __NR_memfd_create
+#  else
+/* These values come from <https://fedora.juszkiewicz.com.pl/syscalls.html>. */
+#    warning "libc is outdated -- using hard-coded SYS_memfd_create"
+#    if defined(__x86_64__) // x86_64
+#      define SYS_memfd_create 319
+#    elif defined(__i386__) // x32
```

Comment:
i386 isn't x32
https://en.wikipedia.org/wiki/X32_ABI

Reply:
Ah, right. The `#ifdef` is right but the comment is wrong.